core/chain_manager_test.go

```diff
@@ -483,22 +485,71 @@ func TestInsertNonceError(t *testing.T) {
 	}
 }
 
-/*
-func TestGenesisMismatch(t *testing.T) {
-	db, _ := ethdb.NewMemDatabase()
-	var mux event.TypeMux
-	genesis := GenesisBlock(0, db)
-	_, err := NewChainManager(genesis, db, db, db, thePow(), &mux)
-	if err != nil {
+func TestChainReorgMissingTransactions(t *testing.T) {
```

Comment:
Hmm, sorry, it was my half ass implementation that failed, however, the test is still wrong in that they are not the same transactions. Maybe it would help to have separate tests to test removal, and override?